contracts/PToken.sol

```diff
@@ -0,0 +1,110 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+import "./PTRC21.sol";
+
+contract PToken is Privacy {
+    constructor (address token,
+        uint256 minFee,
+        uint256 depositFee,
+        uint256 withdrawFee
+    )  PTRC21(token, minFee, depositFee, withdrawFee) public {}
+    
+    function deposit(
+        uint256 value,
+        uint _pubkeyX,
+        uint _pubkeyY,
+        uint _txPubKeyX,
+        uint _txPubKeyY,
+        uint256 _mask,
+        uint256 _amount,
+        uint256 _encodedMask,
+        byte[137] memory _data) public  {
+            uint actualAddAmount = doTransferIn(msg.sender, value);
+            
+            uint256 _value = _toPrivacyValue(actualAddAmount);
+            require(_value > getDepositFee(), "deposit amount must be strictly greater than deposit fee");
+            
+            _deposit(
+                _value,
+                _pubkeyX,
+                _pubkeyY,
+                _txPubKeyX,
+                _txPubKeyY,
+                _mask,
+                _amount,
+                _encodedMask,
+                _data);
+    }
+
+    function transferFee(uint256 fee) internal {
+        uint256 _externalValue = _toExternalValue(fee);
+        ITRC21 _token = ITRC21(token());
+        _token.transfer(issuer(), _externalValue);
```

Comment:
this needs to be very careful as the actual token transferred from the contract would be: the fee of the token (caused by `transfer` function) + `_externalValue`, which results in the sum of the token hold of the contract is less than the sum of all privacy token holders => withdrawals will be failed

Reply:
:+1: